File: php/features.php
<?php
declare(strict_types=1);

// FuelPricesAtSpanishGasStations SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FuelPricesAtSpanishGasStationsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FuelPricesAtSpanishGasStationsBaseFeature();
            case "test":
                return new FuelPricesAtSpanishGasStationsTestFeature();
            default:
                return new FuelPricesAtSpanishGasStationsBaseFeature();
        }
    }
}

File: rb/features.rb
# FuelPricesAtSpanishGasStations SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FuelPricesAtSpanishGasStationsFeatures
  def self.make_feature(name)
    case name
    when "base"
      FuelPricesAtSpanishGasStationsBaseFeature.new
    when "ratelimit"
      FuelPricesAtSpanishGasStationsRatelimitFeature.new
    when "retry"
      FuelPricesAtSpanishGasStationsRetryFeature.new
    when "test"
      FuelPricesAtSpanishGasStationsTestFeature.new
    when "timeout"
      FuelPricesAtSpanishGasStationsTimeoutFeature.new
    else
      FuelPricesAtSpanishGasStationsBaseFeature.new
    end
  end
end
